src/Sarif.Viewer.VisualStudio.Core/Sarif/ArtifactChange.Extensions.cs

```diff
@@ -41,5 +41,28 @@ public static ArtifactChangeModel ToArtifactChangeModel(this ArtifactChange file
 
             return model;
         }
+
+        public static ArtifactChangeModel ToArtifactChangeModel(this ArtifactChange fileChange, IDictionary<string, Uri> originalUriBaseIds, FileRegionsCache fileRegionsCache)
+        {
+            if (fileChange == null)
+            {
+                return null;
+            }
+
+            Dictionary<string, ArtifactLocation> uriToArtifactLocationMap = null;
+
+            if (originalUriBaseIds != null)
+            {
+                // convert IDictionary<string, Uri> to IDictionary<string, ArtifactLocation> which
+                // is needed by Sarif.SDK extension method ArtifactLocation.TryReconstructAbsoluteUri()
+                uriToArtifactLocationMap = new Dictionary<string, ArtifactLocation>(originalUriBaseIds.Count);
+                foreach (KeyValuePair<string, Uri> entry in originalUriBaseIds)
+                {
+                    uriToArtifactLocationMap.Add(entry.Key, new ArtifactLocation { Uri = entry.Value });
+                }
+            }
+
+            return fileChange.ToArtifactChangeModel(uriToArtifactLocationMap, fileRegionsCache);
```

Comment:
>fileChange.ToArtifactChangeModel(uriToArtifactLocationMap, fileRegionsCache) [](http://example.com/codeflow?start=19&length=76)

there is a possibility where originalUriBaseIds == null, with that, uriToArtifactLocationMap would be null.

Can you check if fileChange.ToArtifactChangeModel won't throw an exception, pls?

Reply:
yes fileChange.ToArtifactChangeModel() allows null Dictionary<string, ArtifactLocation> as parameter and code handles that and will not throw exception.

so the change here allows null Dictionary<string, Uri> as parameter of the overload method, only add values if originalUriBaseIds is not null.